play/src/main/scala/caliban/PlayJson.scala

```diff
@@ -0,0 +1,123 @@
+package caliban
+
+import caliban.Value._
+import caliban.parsing.adt.LocationInfo
+import play.api.http.Writeable
+import play.api.libs.json.Reads._
+import play.api.libs.json.Writes._
+import play.api.libs.json._
+
+import scala.util.Try
+
+object PlayJson {
+
+  //// JSON Reads
+  private def jsonToInputValue(json: JsValue): InputValue = json match {
+    case JsNull             => NullValue
+    case boolean: JsBoolean => BooleanValue(boolean.value)
+    case JsNumber(value) =>
+      Try(value.toIntExact).fold(_ => FloatValue(value), IntValue(_))
+    case JsString(value) => StringValue(value)
+    case JsArray(array) =>
+      InputValue.ListValue(array.toList.map(jsonToInputValue))
+    case JsObject(obj) =>
+      val builder = Predef.Map.newBuilder[String, InputValue]
+      builder.sizeHint(obj)
+      obj.foreach { case (k, v) => builder += (k -> jsonToInputValue(v)) }
+      InputValue.ObjectValue(builder.result())
+  }
+
+  implicit val readsInputValue: Reads[InputValue] =
+    json => JsSuccess(jsonToInputValue(json))
+
+  implicit val readsGraphQLRequest: Reads[GraphQLRequest] =
+    for {
+      query         <- (__ \ "query").read[String]
+      operationName <- (__ \ "operationName").readNullable[String]
+      variables     <- (__ \ "variables").readNullable[Map[String, InputValue]]
+    } yield GraphQLRequest(query, operationName, variables)
```

Comment:
This shouldn't be necessary should it? Can't this just be:

```scala
Json.reads[GraphQLRequest]
```

Since all the `Reads` are already present.